rustls/src/msgs/hsjoiner.rs

```diff
@@ -91,32 +91,26 @@ impl HandshakeJoiner {
             None => return Ok(None),
         };
 
-        let msg = parse_message(&self.buf[..len], self.version)?;
+        let buf = &self.buf[..len];
```

Comment:
AIUI the compiler will do stuff like this for us (inline a single-caller, non-pub fn into its call site). I think having single-caller functions often serves a good documentary/readability purpose ("these lines are for parsing a message from a buffer"; a readability hint that was lost here) and ensures calling functions are a bit smaller and more readable.

Reply:
My usual preference is to inline single-caller functions because of the indirection, which I find makes it harder to understand code because you have to skip around to gain context -- as with so many things, it really depends. If the goal is to make it clear what the function of a block of code is, adding some comments can often serve that purpose quite well.

Would you prefer I drop this commit? It's definitely easy to do so? Could also just add a few lines of comments.

Reply:
Added some comments.